packages/@react-aria/combobox/docs/useComboBox.mdx

```diff
@@ -0,0 +1,1171 @@
+      <DismissButton onDismiss={() => state.close()} />
+      <ul
+        {...mergeProps(listBoxProps, otherProps)}
+        ref={listBoxRef}
+        style={{
+          position: 'absolute',
+          width: '100%',
+          margin: '4px 0 0 0',
+          padding: 0,
+          listStyle: 'none',
+          border: '1px solid gray',
+          background: 'lightgray'
+        }}>
+        {[...state.collection].map(item => (
+          <Option
+            shouldUseVirtualFocus
+            key={item.key}
+            item={item}
+            state={state} />
+        ))}
+      </ul>
+      <DismissButton onDismiss={() => state.close()} />
+    </div>
+  )
+}
+
+function Option({item, state, shouldUseVirtualFocus}) {
+  let ref = React.useRef();
+  let isDisabled = state.disabledKeys.has(item.key);
+  let isSelected = state.selectionManager.isSelected(item.key);
+  // Track focus via focusedKey state instead of with focus event listeners
+  // since focus never leaves the text input in a ComboBox
+  let isFocused = state.selectionManager.focusedKey === item.key;
+
+  // Get props for the option element.
+  // Prevent options from receiving browser focus via shouldUseVirtualFocus.
+  let {optionProps} = useOption({
+    key: item.key,
+    isDisabled,
+    isSelected,
+    shouldSelectOnPressUp: true,
+    shouldFocusOnHover: true,
+    shouldUseVirtualFocus
+  }, state, ref);
+
+  let backgroundColor;
+  let color;
+
+  if (isSelected) {
+    backgroundColor = 'blueviolet';
+    color = 'white';
+  } else if (isFocused) {
+    backgroundColor = 'gray';
+    color = 'black';
+  } else if (isDisabled) {
+    backgroundColor = 'transparent';
+    color = 'gray';
+  }
+
+  return (
+    <li
+      {...optionProps}
+      ref={ref}
+      style={{
+        background: backgroundColor,
+        color: color,
+        padding: '2px 5px',
+        outline: 'none',
+        cursor: 'pointer'
+      }}>
+      {item.rendered}
+    </li>
+  );
+}
+
+<ComboBox label="Favorite Animal" defaultSelectedKey="red panda">
+  <Item key="red panda">Red Panda</Item>
+  <Item key="cat">Cat</Item>
+  <Item key="dog">Dog</Item>
+  <Item key="aardvark">Aardvark</Item>
+  <Item key="kangaroo">Kangaroo</Item>
+  <Item key="snake">Snake</Item>
+</ComboBox>
+```
+
+### Controlled
+
+The following example shows how you would create a controlled ComboBox, controlling everything from the selected value (`selectedKey`)
+to the combobox options (`items`). By passing in `inputValue`, `selectedKey`, `isOpen`, and `items` to the `useComboBoxState` hook you can control
+what exactly your ComboBox should display.
+
+It is important to note that you don't have to control every single aspect of a ComboBox. If you decide to only control a single property of the
+ComboBox, be sure to provide the change handler for that prop as well e.g. controlling `selectedKey` would require `onSelectionChange` to be passed to `useComboBox` as well.
+
```

Comment:
Should we expand on the fact that the filtering in the example is different than the previous one precisely because it's controlled?

Reply:
Sure, I'll add a little blurb